### PR TITLE
Bump rails from v6.0 to v7.2 + ruby from v2.7 to v3.3

### PR DESCRIPTION
**Pull request by ksylvest**

### CODE REVIEW

File: app/controllers/api/builds/recommended_versions_controller.rb [hunk 1]
```diff
@@ -9,7 +7,7 @@ def show
           json: {
             current_version: latest_build&.version,
             minimum_supported_version: minimum_supported_build.version,
-            manifest_url: manifest_url
+            manifest_url:
```
Comment (MichaelM1177) [inline app/controllers/api/builds/recommended_versions_controller.rb:10]: is this just a new rails thing, where if the names match, you can omit?

File: app/models/build.rb [hunk 2]
```diff
@@ -87,6 +83,6 @@ def create_deploy
   end
 
   def handle_new_minimum_supported_version
-    Build.where.not(id: id).where(minimum_supported_version: true).update_all(minimum_supported_version: false)
+    Build.where.not(id:).where(minimum_supported_version: true).update_all(minimum_supported_version: false)
```
Comment (MichaelM1177) [inline app/models/build.rb:86]: same Q here

File: app/models/auth.rb [hunk 3]
```diff
@@ -0,0 +1,60 @@
+# Encapsulates the authentication for Google.
+class Auth
+  class CredentialsError < SecurityError; end
+
+  HD = ENV.fetch('GOOGLE_HD', 'clutter.com')
```
Comment (MichaelM1177) [inline app/models/auth.rb:5]: is this keeping it only to `clutter.com`? curious if it can allow both `clutter.com` and `ironmountain.com`

**Review by MichaelM1177 — APPROVED** [review]

File: app/controllers/api/builds/recommended_versions_controller.rb (hunk 1 above)
Comment (ksylvest) [reply app/controllers/api/builds/recommended_versions_controller.rb:10]: Yeah, it is a new ruby 3.1 feature (that rubocop flags) that letes you omit a value if it is named the same as a key for both hashes and keyword-args:

```ruby
n = 5
fibonacci(n:) == fibonacci(n: n)
```

```ruby
y = 5
{ y: } == { y: y}
```

https://dev.to/baweaver/ruby-3-1-shorthand-hash-syntax-first-impressions-19op

File: app/models/build.rb (hunk 2 above)
Comment (ksylvest) [reply app/models/build.rb:86]: Same callout.

File: app/models/auth.rb (hunk 3 above)
Comment (ksylvest) [reply app/models/auth.rb]: Yeah - good callout. Can allow both. I've split that into this commit:

80729699d976161964affad285bb21ceb8936815

The only change is the login screen does not by default display only @clutter.com addreses.

**Pull request merged**